src/main/java/fridgy/model/ingredient/Quantity.java

```diff
@@ -27,20 +28,28 @@
      */
     public Quantity(String quantity) {
         requireNonNull(quantity);
-        AppUtil.checkArgument(isValidQuantity(quantity), MESSAGE_CONSTRAINTS);
-        value = convertToStandardUnit(quantity);
+        AppUtil.checkArgument(isValidQuantityString(quantity), MESSAGE_CONSTRAINTS);
+        String[] results = QuantityCalc.standardiseQuantity(quantity).split(" ");
+        value = Double.parseDouble(results[0]);
+        if (results.length > 1) {
+            units = results[1];
+        } else {
+            units = "";
+        }
```

Comment:
Hide this parsing through splitting string in a method, keep constructor easy to read.
```suggestion
        String standardQuantity = QuantityCalc.standardiseQuantity(quantity);
        this.value = getQuantityValue(standardQuantity);
        this.units = getQuantityUnits(standardQuantity);
```

Reply:
okay adjusted 🐒